### PR TITLE
Week 3 dustin schell

### PR DESCRIPTION
**Pull request by dustin-oz**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,42 +15,173 @@
 
 // a) Create a test with expect statements for each of the variables provided.
 
+// --- > ::STUDENT NOTES:: This question confused me at first because this problem was not supplied with variables like it states in the test question/statement, only suggested example inputs? With that said, I bypassed this statement and proceeded to make the test work as I suspect I am supposed to.
```
Comment (elysemontano) [inline code-challenges.test.js:18]: Yes, you approached this the right way.  Thank you for writing this in here!  If you every need clarification, please feel free to reach out to one of us when you are working on your assessment, we are here to help!

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,42 +15,173 @@
...
+
+const sequencer = (number) => {
+    let array = [1, 1]
+    for (let i = 0; i < number - 2; i++) {
+        if (number > 2) {
+            array.push(array[i] + array[i + 1])
+        }
+    }
+    return array
+}
```
Comment (elysemontano) [inline code-challenges.test.js:64]: This looks great!  Since you took into account an edge case in this code, you are always welcome and even encouraged to test those edge cases with your test as well!

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,42 +15,173 @@
...
-const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
-// Expected output: [-9, 7, 9, 199]
+describe("oddNumberSorter", () => {
+    const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+    const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+    it("takes in an array and returns a new array of only odd numbers sorted from least to greatest", () => {
+        expect(oddNumberSorter(fullArr1)).toEqual([-9, 7, 9, 199])
+        expect(oddNumberSorter(fullArr2)).toEqual([-823, 7, 23])
+    })
+})
```
Comment (elysemontano) [inline code-challenges.test.js:86]: Test looks awesome!

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,42 +15,173 @@
...
+// Return the supplied array and FILTER it based on TYPEOF NUMBER and whether it is ODD NUMBER
+// Return the array and SORT it least to greatest
+
+
+const oddNumberSorter = (array) => {
+    return array.filter(value => {
+        if (value % 2 !== 0 && typeof (value) === "number") {
+            return array
+        }
+    }).sort((alpha, bravo) => alpha - bravo)
```
Comment (elysemontano) [inline code-challenges.test.js:119]: Nice! 😄

File: code-challenges.test.js [hunk 5]
```diff
@@ -15,42 +15,173 @@
...
 
 
+// PSEUDO CODE
+// ----------------------
+// Create a FUNCTION called oddNumberSorter that takes in an ARRAY
+// Return the supplied array and FILTER it based on TYPEOF NUMBER and whether it is ODD NUMBER
+// Return the array and SORT it least to greatest
+
+
+const oddNumberSorter = (array) => {
```
Comment (elysemontano) [inline code-challenges.test.js:114]: Logic looks great on this problem!  A way to simplify this a little further is to remove the conditional.  .filter has a built-in conditional statement, so there is no need to add one in.

File: code-challenges.test.js [hunk 6]
```diff
@@ -15,42 +15,173 @@
...
 const numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+    const numbersToAdd2 = [0, 7, -8, 12]
+    const numbersToAdd3 = []
+    it("takes in an array and returns an array of the accumulating sum. An empty array should return an empty array", () => {
+        expect(arrayAccumulator(numbersToAdd1)).toEqual([2, 6, 51, 60])
+        expect(arrayAccumulator(numbersToAdd2)).toEqual([0, 7, -1, 11])
+        expect(arrayAccumulator(numbersToAdd3)).toEqual([])
+    })
+})
```
Comment (elysemontano) [inline code-challenges.test.js:144]: 💯

File: code-challenges.test.js [hunk 7]
```diff
@@ -15,42 +15,173 @@
...
+const arrayAccumulator = (array) => {
+    let newArray = []
+    return array.map((value, index) => {
+        if (index > 0) {
+            return newArray = value + newArray
+        } else {
+            return newArray = value
+        }
+    })
+}
```
Comment (elysemontano) [inline code-challenges.test.js:181]: Excellent!  Great work 👏

File: interview-questions.md [hunk 8]
```diff
@@ -9,59 +9,63 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: From memory, I believe the required lifecycle method in this case is the constructor.
 
-  Researched answer:
+  Researched answer: After looking up this answer in the syllabus, it appears I was on target with my initial answer.
```
Comment (elysemontano) [inline interview-questions.md:30]: Since React is always looking to display something or _render_ the page, the lifecycle method is the render() method.

File: interview-questions.md [hunk 9]
```diff
@@ -9,59 +9,63 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: If I remember correctly, an anonymous function is a function without a name.
 
-  Researched answer:
+  Researched answer: After researching this answer, it appears I was on target in my initial answer.
```
Comment (elysemontano) [inline interview-questions.md:54]: Good job in this section.  I would recommend seeing if you can expand even further in your researched answers.

File: react-debugging.html [hunk 10]
```diff
@@ -79,6 +80,7 @@ <h1>Draw a Card</h1>
                 )
               })}
             </div>
+            </>
```
Comment (elysemontano) [inline react-debugging.html:83]: Well done!

**Review by elysemontano — APPROVED** [review]
Crushed this assessment Dustin!  Your pseudocode is phenomenal and you have a great approach to the code challenge problems.  I would recommend expanding your interview answers a little further.  The researched answer is not necessarily to verify your accuracy, but to stretch your knowledge further.  Awesome work!
